File: Lab6 Yura Romanchuk/TestFramework/Pages/BasePage.cs
﻿using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace PageObjects.Pages
{
    public class BasePage
    {
        protected IWebDriver _driver;

        public BasePage(IWebDriver driver)
        {
            _driver = driver;
            PageFactory.InitElements(driver, this);
        }
    }
}

File: Lab7_TrokhymovychMykola/Lab7_TrokhymovychMykola/src/main/java/PageObjects/BasePage.java
package PageObjects;

import elements.customdecorator.CustomFieldDecorator;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.PageFactory;

public class BasePage {
    private WebDriver driver;

    public BasePage(WebDriver driver){
        this.driver = driver;
        PageFactory.initElements(new CustomFieldDecorator(driver), this);
    }

    public WebDriver getDriver() {
        return driver;
    }
}
